src/main/scala/tech/cryptonomic/conseil/routes/Tezos.scala

```diff
@@ -62,49 +61,74 @@ object Tezos {
 class Tezos(implicit apiExecutionContext: ExecutionContext) extends LazyLogging with DatabaseApiFiltering {
 
   import Tezos._
+  import JsonUtil.{toJson, JsonString}
 
   /* reuse the same context as the one for ApiOperations calls
    * as long as it doesn't create issues or performance degradation
    */
   override val asyncApiFiltersExecutionContext = apiExecutionContext
 
-  //this automatically accepts any type `T` as content for calling [[RequestContext.complete]]
-  //converts to json string via JsonUtil adding the correct content-type to the response entity
-  implicit def jsonStringMarshaller[T]: ToEntityMarshaller[T] =
+  //add the correct content-type for [[JsonUtil]]-converted values
+  implicit val jsonMarshaller: ToEntityMarshaller[JsonString] =
     PredefinedToEntityMarshallers.StringMarshaller
-      .compose(JsonUtil.toJson[T])
-      .wrap(MediaTypes.`application/json`)(identity)
+      .compose((_: JsonString).json)
+      .wrap(MediaTypes.`application/json`)(identity _)
+
+  /* allow generic handling of optional results, embedded in async computations
+   * in addition to converting any missing result to a NotFound http code, it allows to convert the existing content
+   * to something which is marshallable as a response
+   * @param operation is the computation that will provide, as soon as available, an optional result
+   * @param converter a final conversion function to turn the original T, when available to a marshallable result,
+   *        by default the function converts to a [[JsonString]]
+   * @param T the type of the possible result of the async computation
+   * @param R the final outcome, which must be compatible with an available [[ToResponseMarshaller]]
+   */
+  private[this] def handleNoneAsNotFound[T, R: ToResponseMarshaller](operation: => Future[Option[T]], converter: T => R = toJson[T] _): Future[ToResponseMarshallable] =
```

Comment:
Small note, since you've been using module/object specific private modifiers fairly consistently, it'll probably be better if this is consistent across the entire codebase and any future privates should be module specific. Double check and make sure there are no standalones for this PR.

Reply:
makes sense